bot/cogs/tags.py

```diff
@@ -0,0 +1,265 @@
+import time
+from typing import Optional
+
+from aiohttp import ClientSession
+
+from discord import Colour, Embed
+from discord.ext.commands import AutoShardedBot, Context, command
+
+from bot.constants import ADMIN_ROLE, MODERATOR_ROLE, OWNER_ROLE
+from bot.constants import SITE_API_KEY, SITE_API_TAGS_URL, TAG_COOLDOWN
+from bot.decorators import with_role
+from bot.utils import paginate
+
+
+class Tags:
+    """
+    Save new tags and fetch existing tags.
+    """
+
+    def __init__(self, bot: AutoShardedBot):
+        self.bot = bot
+        self.tag_cooldowns = {}
+
+    async def get_tag_data(self, tag_name: Optional[str] = None) -> dict:
+        """
+        Retrieve the tag_data from our API
+
+        :param tag_name: the tag to retrieve
+        :return:
+        if tag_name was provided, returns a dict with tag data.
+        if not, returns a list of dicts with all tag data.
+
+        """
+        headers = {"X-API-KEY": SITE_API_KEY}
```

Comment:
You're assigning this headers dict with the same value in 3 different functions.
Maybe move it outside of the function scope?